src/itemset.js

```diff
@@ -375,16 +373,12 @@ define([
         return {value: nodeset, filter: ''};
     }
 
-    function getDataSources() {
-        // HACK synchronously get asynchronously loaded data (if available)
-        var sources = [];
-        datasources.getDataSources(function (data) {
-            // will execute synchronously if data sources have been loaded
-            if (opts.dataSourcesFilter) {
-                data = opts.dataSourcesFilter(data);
-            }
-            sources = data;
-        });
+    function getDataSources(mug) {
```

Comment:
I thought about that, but decided against it because it adds duplication at every call site (`mug` -> `mug.form.vellum`). I suppose you could argue that [line 183](https://github.com/dimagi/Vellum/pull/759/files/00af9dd296c945644536b345d36c205ec6927a5c#diff-f15dee15f58a804357f7fca0280de97dR183) could have been simplified since `form` is local there, but it's not even used on [line 179](https://github.com/dimagi/Vellum/pull/759/files/00af9dd296c945644536b345d36c205ec6927a5c#diff-f15dee15f58a804357f7fca0280de97dR179) and I was hesitant to let the scope of changes in this PR creep too far.